wp-web-push/wp-web-push-main.php

```diff
@@ -89,6 +95,44 @@ public static function on_transition_post_status($new_status, $old_status, $post
       }
     }
   }
+
+  public static function get_trigger_by_key_value($key, $value) {
+    foreach(self::$ALLOWED_TRIGGERS as $trigger) {
+      if($trigger[$key] === $value) {
+        return $trigger;
+      }
+    }
+    return false;
+  }
+
+  public static function get_triggers_by_key_value($key, $value) {
+    $matches = array();
+
+    foreach(self::$ALLOWED_TRIGGERS as $trigger) {
+      if($trigger[$key] === $value) {
+        $matches[] = $trigger;
+      }
+    }
+    return $matches;
+  }
+
+  private static function add_trigger_handlers() {
+    $enabled_triggers = get_option('webpush_triggers');
+    foreach($enabled_triggers as $trigger) {
+        $trigger_detail = self::get_trigger_by_key_value('key', $trigger);
+        if($trigger_detail) {
+          add_action($trigger_detail['hook'], array($this, $trigger_detail['action']), 10, 3);
```

Comment:
The problem here is that `array($this, $trigger_detail['action'])` isn't being found by WordPress.  If I make `send_new_post` a global function (i.e. not in the class), and use `$trigger_detail['action']` as the second argument instead of the array, this all works.

Still searching for a solution.